modules/yuktamediaAnalyticsAdapter.js

```diff
@@ -3,134 +3,248 @@ import adapter from '../src/AnalyticsAdapter.js';
-    });
-  } else if (status == 'win') {
-    return {
-      bidderCode: bidResponse.bidder,
-      bidId: bidResponse.requestId,
-      adUnitCode: bidResponse.adUnitCode,
-      auctionId: bidResponse.auctionId,
-      creativeId: bidResponse.creativeId,
-      transactionId: bidResponse.transactionId,
-      currency: bidResponse.currency,
-      cpm: bidResponse.cpm,
-      netRevenue: bidResponse.netRevenue,
-      renderedSize: bidResponse.size,
-      mediaType: bidResponse.mediaType,
-      statusMessage: bidResponse.statusMessage,
-      status: bidResponse.status,
-      renderStatus: 4,
-      timeToRespond: bidResponse.timeToRespond,
-      requestTimestamp: bidResponse.requestTimestamp,
-      responseTimestamp: bidResponse.responseTimestamp
-    }
-  }
+function updateSessionIdTimeout() {
+  storage.setDataInLocalStorage(localStoragePrefix.concat('session_timeout'), Date.now());
+}
+
+function isSessionIdTimeoutExpired() {
+  let cpmSessionTimestamp = storage.getDataFromLocalStorage(localStoragePrefix.concat('session_timeout'));
+  return Date.now() - cpmSessionTimestamp > 3600000;
+}
+
+function getSessionId() {
+  return storage.getDataFromLocalStorage(localStoragePrefix.concat('session_id')) ? storage.getDataFromLocalStorage(localStoragePrefix.concat('session_id')) : '';
+}
+
+function isUtmTimeoutExpired() {
+  let utmTimestamp = storage.getDataFromLocalStorage(localStoragePrefix.concat('utm_timeout'));
+  return (Date.now() - utmTimestamp) > 3600000;
 }
 
 function send(data, status) {
-  let location = utils.getWindowLocation();
-  if (typeof data !== 'undefined' && typeof data.auctionInit !== 'undefined') {
-    data.auctionInit = Object.assign({ host: location.host, path: location.pathname, search: location.search }, data.auctionInit);
-  }
-  data.initOptions = initOptions;
+  const location = utils.getWindowLocation();
+  data.initOptions = Object.assign({ host: location.host, path: location.pathname, search: location.search }, initOptions);
 
-  let yuktamediaAnalyticsRequestUrl = utils.buildUrl({
+  const yuktamediaAnalyticsRequestUrl = utils.buildUrl({
     protocol: 'https',
     hostname: 'analytics-prebid.yuktamedia.com',
-    pathname: status == 'auctionEnd' ? '/api/bids' : '/api/bid/won',
+    pathname: '/api/bids',
     search: {
       auctionTimestamp: auctionTimestamp,
       yuktamediaAnalyticsVersion: yuktamediaAnalyticsVersion,
       prebidVersion: $$PREBID_GLOBAL$$.version
     }
   });
 
-  ajax(yuktamediaAnalyticsRequestUrl, undefined, JSON.stringify(data), { method: 'POST', contentType: 'text/plain' });
+  if (isNavigatorSendBeaconSupported()) {
+    window.navigator.sendBeacon(yuktamediaAnalyticsRequestUrl, JSON.stringify(data));
+  } else {
+    ajax(yuktamediaAnalyticsRequestUrl, undefined, JSON.stringify(data), { method: 'POST', contentType: 'text/plain' });
+  }
 }
 
+var yuktamediaAnalyticsAdapter = Object.assign(adapter(
+  {
+    emptyUrl,
```

Comment:
Why pass an empty url? If you are overriding the track function, I don't think you need to include this property as an argument to the adapter interface call.

Reply:
removed emptyUrl. Thanks for improving.